background/imports.js

```diff
@@ -1,141 +1,166 @@
-                               { 'aliases': new RegExp(line[2], 'i') }
-                            ]
-                        }).exec(function(err, category){
-                            if(err) {console.log(err);}
-                            if(category){
-                                Torrent.findOne({
-                                    infoHash: line[0]
-                                }).exec(function(err, exists){
-                                    if(!exists){
-                                        Torrent.create({
-                                            title: line[1],
-                                            category: category._id,
-                                            size: line[5],
-                                            details: [
-                                                line[3]
-                                            ],
-                                            swarm: {
-                                                seeders: line[8],
-                                                leechers: line[9]
-                                            },
-                                            lastmod: Date.now(),
-                                            imported: Date.now(),
-                                            infoHash: line[0]
-                                        }, function(err) {
-                                            if(err) { console.log(err); }
-                                        });
-                                    }
-                                });
-                            }
-                        });
-                    });
-                }).on('error', function(err) {
-                    console.log(err);
-                }).on('end', function() {
+const cluster = require('cluster');
+const numCPUs = require('os').cpus().length;
+var providers = [];
+var workers = [];
 
-                });
-            } else {
-                console.log('Kat API key is invalid');
+var loadProvider = function (provider) {
+    if(provider) {
+      this.provider = provider;
+    }
+    log.info('Loading provider ' + this.provider);
+    try {
+        require(__dirname + '/providers/' + this.provider);
+    } catch (e) {
+        log.info('An error occurred loading the ' + this.provider + ' provider.');
+        log.info('The error is as follows:');
+        log.info(e);
+        console.trace(e);
+    }
+};
+
+/**
+ * The idea here is to use the Node.js {@link https://nodejs.org/api/cluster.html|cluster} module
+ * to take advantage of multi-core systems (i.e every modern computer). Node.js is single threaded
+ * so the cluster module spawns multiple node.js processes. This should theoretically allow us to
+ * process multiple archives at a time. However if this code is io bound rather than CPU bound it
+ * might make no difference at all. Further testing is required.
+ */
+if(cluster.isMaster) {
```

Comment:
This may all need to be removed as we should be moving to using ES6 and imports instead of require.